src/IO/ForkWriteBuffer.cpp

```diff
@@ -0,0 +1,83 @@
+#include <IO/ForkWriteBuffer.h>
+#include <Common/Exception.h>
+#include <ranges>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+extern const int CANNOT_CREATE_IO_BUFFER;
+}
+
+ForkWriteBuffer::ForkWriteBuffer(WriteBufferPtrs && sources_)
+    : WriteBuffer(nullptr, 0), sources(std::move(sources_))
+{
+    if (sources.empty())
+    {
+        first_buffer = nullptr;
```

Comment:
this assignment is redundant.

Reply:
addressed through the other comment by removing `first_buffer`